### ISSUE TITLE
Extend item coupling

### ISSUE THREAD

**Issue by cenxun** [LeaVerou/awesomplete#16883]
sorry, my English is not very good.
i hope have a item display to handle the function, but does not involve the creation of LI, just return Li data 

as follows：

``` javascript
item: function(text, input) {
    // in my code, the text is object, i need to convert string
    var title = text.label.liName;
    var html = input === '' ? text : text.replace(RegExp($.regExpEscape(input.trim()), "gi"), "<mark>$&</mark>");
    return {
        innerHTML: html,
        "aria-selected": "false"
    };
}
```

and i don't have to care how to generate li dom, hope to separate the data and the view.
Just a suggestion.
Thank you very much for providing such a good plugin.

**Comment by LeaVerou** [comment]
@vlazar, isn't that what the `data` function does? If I'm understanding this question correctly (it was rather hard to parse).

**Comment by vlazar** [comment]
@LeaVerou you are absolutely right.

If @cenxun you question was about changing item data without providing a custom `item` function (which is a render function), then you can use `data` function to modify original item data.

Each item in Awesomplete can have a `label` and `value` if needed. For simple case with items as strings internal item `label` and `value` are the same.

Here is an example of how to add `"Modified "` string for each item label:

If original items list is an array of strings.

``` javascript
data: function (item, input) {
    return "Modified " + item;
}
```

If original items list is an array of objects with `label` and `value` keys.

``` javascript
data: function (item, input) {
    return { label: "Modified " +  item.label, value: item.value };
}
```

If original items list is an array of arrays where first element in item array is `label` and the second is `value`.

``` javascript
data: function (item, input) {
    return [ "Modified " +  item[0], item[1] ];
}
```

See docs about `data` function in this section http://leaverou.github.io/awesomplete/#extensibility.
And email autocomplete example here http://leaverou.github.io/awesomplete/#advanced-examples.

**Comment by LeaVerou** [comment]
Thanks @vlazar. Closing this now, since @cenxun hasn't replied for 5 days.

**Comment by ghost** [comment]
WTF? I unsubscribed and all that horseshit.  Apple really needs to create a 'block email' feature so I don't get harassed by your incessant messages about shit I do not care about.  Please make it stop.

Thank you,
Stacey

**Comment by LeaVerou** [comment]
Sigh. You need to unsubscribe from the entire repo. You probably unsubscribed from that one discussion. See this button? 
![image](https://cloud.githubusercontent.com/assets/175836/14765417/672ee544-09ae-11e6-84ed-5fcc3b1f1283.png)
Click it and select "Not watching".

Also, you might need to work on your manners. Nobody here is responsible for whoever subscribed to this repo with your email. Direct the rudeness to whoever 20 year old in your family registered here with your email.

**Comment by meganatas** [comment]
No such thing happened
Just want what is fair to me and all the other hard working people you do
this to I know I'm not alone in this
justin

**Comment by meganatas** [comment]
I'm not doing any off that until I know this ain't just another one of your
scams I've kept every text every chat  all of it I will get to the bottom
of this count on it and for my additude you take my money rip  me off and
I'm supposed to smile and be happy

justin

**Comment by TxHawks** [comment]
Wow, some people are wacky...

**Comment by ghost** [comment]
My manners are fine thank you.   And thank you for providing proper instructions on how to 'unsubscribe', on this my third attempt.  Most sites make the  'unsubscribe' process simple, mostly because they're required to.  GitHub does not, as clearly I am not the first person to attempt to unsubscribe only to be harassed more for my efforts. 

I signed up for the site.  You should worry more about your grammar, your ability to understand an issue and your ability to construct a point than others' manners.   

Thank you,
Stacey

**Comment by cenxun** [comment]
@LeaVerou @vlazar, sorry, I didn't say that clearly.
Because I want to rewrite `filter` method, like is 

``` javascript
filter: function (text, input) {
    var a = text.label.acronym.indexOf(text),
        b = text.label.pinyin.indexOf(text);
    return b >= 0 || a >= 0;
}
```

 i can't use `data` method,  cause I modify the `data` method will be lost to other fields(i need these fields to filter )
At the moment, I'm using it.

``` javascript
item : function(text, input) {
    var li = document.createElement('li');
    li.setAttribute('aria-selected', false);
    li.textContent = text.label.Fname;
    return li;
},
```

so put this proposal . just a suggestion.
Thanks.
